src/image.rs

```diff
@@ -617,6 +618,22 @@ pub trait ImageDecoder<'a>: Sized {
 
         Ok(())
     }
+
+    /// Set decoding limits for this decoder. See [`Limits`] for the different kinds of 
+    /// limits that is possible to set.
+    ///
+    /// Note to implementors: make sure you call [`Limits::check_support`] so that 
+    /// decoding fails if any unsupported strict limits are set.
+    /// [`Limits::check_support`] will also check the `max_image_width` and
+    /// `max_image_height` limits.
+    ///
+    /// [`Limits`]: ./io/struct.Limits.html
+    /// [`Limits::check_support`]: ./io/struct.Limits.html#method.check_support
+    fn set_limits(&mut self, limits: crate::io::Limits) -> ImageResult<()> {
+        limits.check_support(self, crate::io::LimitSupport::default())?;
+
```

Comment:
This could/should also check the width and height, right?

Reply:
`Limits::check_support` takes the decoder by reference so that it can check width and height. I thought this would be simpler as this is a check all decoders would have to do.

Reply:
Ah, I see. Didn't realize that from the name